isisd/isis_nb_config.c

```diff
@@ -1848,9 +1848,6 @@ int isis_instance_mpls_ldp_sync_create(struct nb_cb_create_args *args)
 
 	switch (args->event) {
 	case NB_EV_VALIDATE:
-		if (isis == NULL)
-			return NB_ERR_VALIDATION;
-		break;
```

Comment:
fixed

Reply:
unless something changed in the NB API since the last time I touched it, which arguably was a long time ago, we should never be returning validation errors in the APPLY phase - they should have been caught earlier (@rwestphal  correct me if I am wrong). 

I am not familiar with how the multi-VRF support was implemented, but at a glance it appears we are still creating the global isis instance for the default VRF when the daemon is started, in isis_main.c:
```
  263         /* create the global 'isis' instance */                                 
  264         isis_global_instance_create(VRF_DEFAULT_NAME);   
```
so I think it is safe to assert. In any case the original fix looks more correct than the amended one.

Reply:
I think this is correct ... it should be safe to `assert` here

Reply:
@edipascale : Yes you are correct, we should avoid to return validation errors from APPLY phase.
But in some cases it's required when daemon has not enough information during the VALIDATION phase.

If isis with non default vrf gets configured, then it is getting created in API isis_instance_create() in apply phase.
If two configurations isis instance and ldp sync coming together(in one transaction).
In case of transactional mode, the sequence of API calls would be
isis_instance_create() and then isis_instance_mpls_ldp_sync_create() for validation phase. 
I think Quintin is pointing to this use case.

Here It looks isis_instance_mpls_ldp_sync_create() has a support for default vrf and default isis only. Although default vrf, isis check would not fail at present, but later It can.

I feel Quintin is correct to perform the check in APPLY phase.

Reply:
> unless something changed in the NB API since the last time I touched it, which arguably was a long time ago, we should never be returning validation errors in the APPLY phase - they should have been caught earlier 

Yes, hence my suggestion to simply assert in APPLY if we know `isis` should never be null there.

Reply:
fixed